tests/preprocess.py

```diff
@@ -214,6 +219,94 @@ def test_process_include_file_with_multiple_files():
     assert defines['CONST3'] == '777', "constant from incl2.h"
 
 
+@test
+def test_process_include_file_using_database():
```

Comment:
did you try running these on the esp32 device?

Reply:
I have not run the unit tests on the device. I have run all .S files from the compat tests and from the tests I added as compat_rtc_tests on the device though and they worked as expected. (Actually.. I did not then download the resulting binary to compare with binutils - I guess under the assumption that MicroPython on or off the device would run the same.)

What is the reason you are asking? The know that the unit tests can run on the device? Or to ensure it all works in the limited memory (which my testing have shown so far), or something else? Depending on what you were after, I can do more testing.

Reply:
Btw, just ran all unit test on the esp32 and they all pass.

Reply:
Yay, great.

I was asking because running them on esp32 might show more issues compared to running it on dev PC.